lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -823,15 +862,51 @@ def upload_reads(self, ctx, params):
         self.log('Starting upload reads, parsing args')
         o, wsid, name, objid, kbtype, single_end, fwdid, revid, shock = (
             self._proc_upload_reads_params(params))
-        interleaved = 1 if (not single_end and not revid) else 0
+        #IF from shock fwdid and revid are shock nodes, if not shock they are file paths
+        dfu = DataFileUtil(self.callback_url)
+        fwdpath = None
+        revpath = None
         if shock:
-            fhandle, rhandle, fsize, rsize = self.process_shock_ids_for_upload(
-                fwdid, revid, interleaved)
+            #Grab files from Shock
+            fileinput = [{'shock_id': fwdid,
+                          'file_path': self.scratch + '/fwd/',
+                          'unpack': 'uncompress'}]
+            if revid:
+                fileinput.append({'shock_id': revid,
+                                  'file_path': self.scratch + '/rev/',
+                                  'unpack': 'uncompress'})
+            self.log('downloading reads file(s) from Shock')
+            files = dfu.shock_to_file_mass(fileinput)
+            shock = False
+            fwdpath = files[0]["file_path"]
+            fwdname = files[0]["node_file_name"]
+            if revid:
+                revpath = files[1]["file_path"]
+                revname = files[1]["node_file_name"]
         else:
-            fhandle, rhandle, fsize, rsize = self.process_files_for_upload(
-                fwdid, revid, interleaved)
+            #Not shock
+            fwdpath = fwdid
+            revpath = revid
+            fwdname = None
+            revname = None
+            fwdid = None
+            revid = None
+        if revpath:
+            #now interleave the files
+            intpath = os.path.join(self.scratch, self.get_file_prefix() + '.inter.fastq')
+            self.interleave(None, None, fwdname, fwdid,
+                            revname, revid, fwdpath, revpath, intpath)
+            fwdpath = intpath
+            revpath = None
+
+        interleaved = 1 if not single_end else 0
+
+        fhandle, rhandle, fsize, rsize = \
+            self.process_files_for_upload(fwdpath, revpath, interleaved)
```

Comment:
Increasing maintenance burden and code complexity for something that might happen in the future is not a reasonable approach. Always simplify if you can. Especially in this case since the now unnecessary more complex code is in the git history.